### ISSUE TITLE
[feature-request]: Implement habitat as a field in class Event

### ISSUE THREAD

**Issue by mpitblado** [gbif/gbif-api#131]
This request pertains to http://rs.tdwg.org/dwc/terms/habitat

Upon looking in src/main/java/org/gbif/api/model/event/Event.java and https://techdocs.gbif.org/en/openapi/v1/occurrence#/Searching%20occurrences/searchOccurrence it does not seem like habitat is a currently available field through the api, while it is supported by the ipt. I am submitting this request as we have users that have expressed a strong interest in being able to use this to query for records within their datasets.

Habitat is not available as an option in either the main gbif.org UI or the hosted portal UI. I am not sure if that would also need to be implemented if this request were to be accepted to complete the overall objective, or if those UI's pull all available fields exposed through the API automatically.

**Comment by CecSve** [comment]
One of the challenges with implementing search on the dwc:habitat field is that the values are not standardized (no controlled vocabulary exist), and implementing one would be difficult although we do have an open [issue](https://github.com/gbif/vocabulary/issues/75) to do so. 

Currently there are more than 13 million verbatim values in the habitat field and values tend to only be present in one dataset. The verbatim values are usually long-ish text strings, for example:
`
'                            Along dry stream running more or less South out of range. Sandstone.'`
`'     Insufficient water/flow available to collect a field water quality measurement.'`
`'   Insufficient water/flow available to collect a field water quality measurement.'`
`'A medium straight tree with unbuttressed bole, broad crown. Height 80 ft [24.5 m]. Bole 35 ft [10.5 m]. d.b.h. 20 ins'`
`'  A small malformed, many stemmed tree, 20 feet [6 m] total height. Bark 1/8 thick, outer bark smooth, few large'`
`'Bastard' softwood scrub.'`

> I am submitting this request as we have users that have expressed a strong interest in being able to use this to query for records within their datasets.

Just so I am sure I understand you correctly - your user community would like to be able to filter their data based on habitat prior to downloading the data? 

It is possible to get habitat data from the verbatim file of a full DwC-A download: https://techdocs.gbif.org/en/data-use/download-formats#dwca-verbatim, but that of course does not allow users to search for specific habitats.

Unless the field is standardized by a controlled vocabulary, I doubt it would make much sense to enable a search filter.

**Comment by mpitblado** [comment]
Hi CecSve,

Correct, our user community would like to be able to search records within a dataset based on a pattern for the field habitat. For example, 

| Habitat |
| ------- |
| Long Beach |
| Test Rock |
| Null Island |

Could be queried on `*Island*` to return

| Habitat |
| ------- |
| Null Island |

Ultimately, my objective is to get the term included in the hosted portal interface, but my understanding is that before a term gets on the list of filters, it must first be available within the API, hence this request.

![image](https://github.com/gbif/gbif-api/assets/116753126/6d462a9a-5ae4-45c0-9f42-d7b01293412f)

On gbif.org, a user cannot also currently search via habitat. Again my understanding is that in order to become available in that UI, it would also need to first be implemented through the API.

![image](https://github.com/gbif/gbif-api/assets/116753126/8599c266-6f6e-497b-96b0-bc9b2d58d99f)

The intended search functionality would be identical to other non-standardized text fields, such as recordNumber, locality, waterBody etc. However, if habitat is not commonly queued and the compute required is judged as excessive, I understand.

**Comment by CecSve** [comment]
> On gbif.org, a user cannot also currently search via habitat. Again my understanding is that in order to become available in that UI, it would also need to first be implemented through the API.

Yes, the field would have to be implemented in he back-end (API) so it can be used by our front-end and a hosted portal.

@MortenHofft could we enable searches on habitat? @mpitblado correct me if I am wrong, but it has relevance for the [The Beaty Biodiversity Museum](https://collections.beatymuseum.ubc.ca/) hosted portal and it is mentioned several times in the upcoming [freshwater publishing guide](https://github.com/gbif/doc-freshwater-data-publishing-guide/issues/13).

**Comment by MortenHofft** [comment]
Once in the API it is a small task to add in the UI. As I understand it it will behave similar to `Verbatim scientific name`.

**Comment by mpitblado** [comment]
Yes @CecSve that is correct, it definitely has relevance to some of the staff/users at the Beaty Biodiversity Museum, and perhaps others who will appreciate the option to search on it once available. As Morten mentions, a standard text search like `Verbatim scientific name` would be great and sufficient, does not need anything special beyond that.
